jvm-libs/linea/blob-decompressor/src/test/kotlin/net/consensys/linea/blob/GoNativeBlobDecompressorTest.kt

```diff
@@ -0,0 +1,42 @@
+package net.consensys.linea.blob
+
+import net.consensys.linea.nativecompressor.CompressorTestData
+import org.assertj.core.api.Assertions.assertThat
+import org.junit.jupiter.api.Assertions.assertTrue
+import org.junit.jupiter.api.BeforeEach
+import org.junit.jupiter.api.Test
+
+class GoNativeBlobDecompressorTest {
+  private val blobCompressedLimit = 10 * 1024
+  private lateinit var compressor: GoNativeBlobCompressor
+  private lateinit var decompressor: BlobDecompressor
+
+  @BeforeEach
+  fun beforeEach() {
+    compressor = GoNativeBlobCompressorFactory
+      .getInstance(BlobCompressorVersion.V1_0_1)
+      .apply {
+        Init(
+          dataLimit = blobCompressedLimit,
+          dictPath = GoNativeBlobCompressorFactory.dictionaryPath.toAbsolutePath().toString()
+        )
+        Reset()
+      }
+    decompressor = GoNativeBlobDecompressorFactory.getInstance(BlobDecompressorVersion.V1_1_0)
+  }
+
+  @Test
+  fun `when blocks are compressed with compressor shall decompress them back`() {
+    val blocks = CompressorTestData.blocksRlpEncoded
+    assertTrue(compressor.Write(blocks[0], blocks[0].size))
+    assertTrue(compressor.Write(blocks[1], blocks[1].size))
+
+    val compressedData = ByteArray(compressor.Len())
+    compressor.Bytes(compressedData)
+
+    val decompressedBlob = decompressor.decompress(compressedData)
+    assertThat(decompressedBlob.size).isGreaterThan(compressedData.size)
+    val decompressedBlocks: List<ByteArray> = rlpDecodeAsListOfBytes(decompressedBlob)
+    assertThat(decompressedBlocks).hasSize(2)
```

Comment:
Why don't we compare the decompressed items to original ones to check if compression - decompression mutates the original data?

Reply:
> to check if compression - decompression mutates the original data?

Our compression is lossy, therefore that is not possible to recover fully the same data.

Want later on to extend the test with deserialization of the RLP and compare if the date is correct, but for that need changes in the Besu classes to allow decoding without signature verification.